config.py

```diff
@@ -1299,3 +1308,37 @@ def initialize_logging(logger_name):
     # re-get the log after logging is initialized
     global log
     log = logging.getLogger(__name__)
+
+def setup_temp_dir(agentConfig, proc_name=None):
```

Comment:
The rationale for keeping it this way is because we don't want to have to handle the branching config logic outside the function. It makes sense to keep it contained here so we don't have to repeat ourselves 4 times. However, it also does make sense to make the function signature clearer.

I think that it's better to have it contained, but if you can think of a way to make it better I'm down to hear that